moveit_setup_assistant/include/moveit/setup_assistant/tools/moveit_config_data.h

```diff
@@ -74,6 +74,7 @@ struct GroupMetaData
   double kinematics_solver_search_resolution_;  // resolution to use with solver
   double kinematics_solver_timeout_;            // solver timeout
   int kinematics_solver_attempts_;              // solver attempts
+  std::string default_planner_;                 // Name of the default planner to use
```

Comment:
I think, instead of saving a copy of the default planner here, you should store a reference into the YAML document structure, which then allows to access all details.